File: LightRunnerLibGDX/src/com/RedmondRNDLabs/lightrunnerlibgdx/Sprite2.java
package com.RedmondRNDLabs.lightrunnerlibgdx;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.math.Rectangle;
import com.badlogic.gdx.math.Vector2;
import com.badlogic.gdx.ApplicationListener;
import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.GL10;
import com.badlogic.gdx.graphics.OrthographicCamera;
import com.badlogic.gdx.graphics.Texture.TextureFilter;
import com.badlogic.gdx.graphics.g2d.Sprite;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.graphics.g2d.TextureRegion;

public class Sprite2 {
	Vector2 Position = new Vector2();
	Rectangle bounds = new Rectangle();
	Texture texture;
	String asset;

	public Sprite2(Vector2 Position, int newW, int newH, String newAsset) {
		this.Position = Position;
		this.bounds.width = newW;
		this.bounds.height = newH;
		this.asset = newAsset;
	}

	public Sprite2(float x, float y, int newW, int newH, String newAsset) {
		this(new Vector2(x, y), newW, newH, newAsset);
	}

	public Vector2 getCenter() {
		return new Vector2(Position.x + bounds.width / 2, Position.y
				+ bounds.height / 2);
	}

	public void setCenter(float centerX, float centerY) {
		Position.x = centerX - bounds.width / 2;
		Position.y = centerY - bounds.height / 2;
	}

	public void setCenterY(float centerY) {
		Position.y = centerY - bounds.height / 2;
	}

	public void loadContent() {
		texture = new Texture(Gdx.files.internal(asset));
	}

	public void draw(SpriteBatch batch) {
		batch.draw(texture, Position.x, Position.y, bounds.width, bounds.height);
	}
}

File: LightRunner-wp7/LightRunner-wp7/LightRunner-wp7/Sprite2.cs
using System;
using System.Collections.Generic;
using System.Linq;
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Audio;
using Microsoft.Xna.Framework.Content;
using Microsoft.Xna.Framework.GamerServices;
using Microsoft.Xna.Framework.Graphics;
using Microsoft.Xna.Framework.Input;
using Microsoft.Xna.Framework.Input.Touch;
using Microsoft.Xna.Framework.Media;
using Microsoft.Phone.Controls;

namespace LightRunner_wp7
{
    public class Sprite2
    {
        public Vector2 position = new Vector2();
        public Vector2 velocity;
        public Rectangle bounds = new Rectangle();
        Rectangle boundingRect = new Rectangle();
        //Polygon p;
        /// <summary>
        /// Rectangular vertices.
        /// </summary>
        float[] vertices = new float[8];
        public Texture2D texture;
        public String asset;

        public Sprite2(Vector2 Position, int newW, int newH)
        {
            this.position = Position;
            this.bounds.Width = newW;
            this.bounds.Height = newH;

            vertices[0] = Position.X;
            vertices[1] = Position.Y;
            vertices[2] = Position.X + bounds.Width;
            vertices[3] = Position.Y;
            vertices[4] = Position.X + bounds.Width;
            vertices[5] = Position.Y + bounds.Height;
            vertices[6] = Position.X;
            vertices[7] = Position.Y + bounds.Height;

            //p = new Polygon(vertices);
        }
        public Sprite2(Vector2 Position, int newW, int newH, String newAsset)
            : this(Position, newW, newH)
        {
            asset = newAsset;
        }
        public Sprite2(float x, float y, int newW, int newH, String newAsset)
            : this(new Vector2(x, y), newW, newH, newAsset)
        {
        }

        public Vector2 getCenter()
        {
            return new Vector2(position.X + bounds.Width / 2, position.Y
                    + bounds.Height / 2);
        }

        public void setCenter(float centerX, float centerY)
        {
            position.X = centerX - bounds.Width / 2;
            position.Y = centerY - bounds.Height / 2;
        }

        public void setCenter(Vector2 newCenter)
        {
            setCenter(newCenter.X, newCenter.Y);
        }

        public void setCenterY(float centerY)
        {
            position.Y = centerY - bounds.Height / 2;
        }

        public void setCenterX(float centerX)
        {
            position.X = centerX - bounds.Width / 2;
        }

        public void updateVertices()
        { }

        public void loadContent(ContentManager contentManager)
        {
            texture = contentManager.Load<Texture2D>("sample");
            bounds.Height = texture.Height;
            bounds.Width = texture.Width;
            updateVertices();
        }

        public void draw(SpriteBatch batch)
        {
            batch.Draw(texture, position, Color.White);
        }
    }
}
